apps/sms/js/contacts.js

```diff
@@ -98,29 +98,26 @@
     findBy: function contacts_findBy(filter, callback) {
       var lower = [];
       var filterValue = (filter.filterValue || '').trim();
-      var terms, request;
+      var terms;
 
       if (!navigator.mozContacts || !filterValue.length) {
-        setTimeout(function() {
-          callback(
-            /**
-             * Bailout Strategy
-             *
-             * 1. A _missing_ filter.filterValue will results in an
-             *    error when passed to mozContacts.find, bailing out
-             *    early avoids the additional overhead of making a
-             *    request that we know will fail.
-             *
-             * 2. Missing navigator.mozContacts API or empty but
-             *    present filter.filterValue make the app useless,
-             *    so prevent unpredictable state by making the
-             *    result a no-op.
-             *
-             */
-            typeof filter.filterValue === 'undefined' ? null : [], {}
-          );
-        });
-        return;
+        /**
+         * Bailout Strategy
+         *
+         * 1. A _missing_ filter.filterValue will results in an
+         *    error when passed to mozContacts.find, bailing out
+         *    early avoids the additional overhead of making a
+         *    request that we know will fail.
+         *
+         * 2. Missing navigator.mozContacts API or empty but
+         *    present filter.filterValue make the app useless,
+         *    so prevent unpredictable state by making the
+         *    result a no-op.
+         *
+         */
+        return Promise.resolve(
```

Comment:
I agree but i'd rather do this in a separate patch because we can clean up a lot of places and I think it's not really part of the migration to promises.